### ISSUE TITLE
Accessibility Bug with Submenus

### ISSUE THREAD

**Issue by chigkim** [surge-synthesizer/surge#8349]
Hi,

Thank you for the great accessibility support.
I use VoiceOver, and the main issue I encounter involves making selections in popup menus. Here are a couple of examples:

1. When an oscillator is set to wavetable, opening the menu to choose a wavetable does not apply the selection after choosing one.
2. When opening a contextual menu (VO+Shift+M) and attempting to add a mod with Add Modulation  From, it does not appear in the modulation list (opt+m). That said, if I do it from the modulation list by select source and select target, it works.

Also, when navigating menus with the up and down arrows, the cursor automatically enters submenus until there are no further levels. I then have to back out using the left arrow.
This occurs frequently when cursor tracking is enabled (VO+Shift+F3) and the mouse pointer is set to follow the VoiceOver cursor in VoiceOver Utilities (VO+F8) > Navigation.
It seems this behavior may be caused by submenus opening automatically on mouse hover? If so, would it be possible to include an accessibility option to disable this behavior, so submenus only open on mouse click or when pressing the right arrow?

Thank you.

MacOS: 26.4.1
Logic: 12.2
Version: Surge XT 1.4.nightly.b5d2d77
Build Info: 2026-04-10 @ 14:20:48 on 'sjc22-be109-f058d4f1-2476-462b-9f09-32cac12dafc9-7E02ACF209CE.local/pipeline' with 'AppleClang-17.0.0.17000013' using JUCE 8.0.10
System Info: macOS 64-bit AU on Apple M3 Max, 64 GB RAM
Host: Apple Logic @ 48000 Hz
Processing Block: 32 samples

**Comment by baconpaul** [comment]
Pretty sure this and #6582 are the same

**Comment by mkruselj** [comment]
Yeah looks like it. Closing this one as a duplicate.

**Comment by chigkim** [comment]
Submenu automatically expanding seems deuplicate  but not 1 and 2. Unless they're related?

1. When an oscillator is set to wavetable, opening the menu to choose a wavetable does not apply the selection after choosing one.
2. When opening a contextual menu (VO+Shift+M) and attempting to add a mod with Add Modulation From, it does not appear in the modulation list (opt+m). That said, if I do it from the modulation list by select source and select target, it works.

**Comment by baconpaul** [comment]
yes you are correct there are 3 separate issues in this issue.

I think 1 is because of the new wavetable widgets. 

I'll re-open the issue.

**Comment by nuoun** [comment]
Is point 1 still happening in the current Nightly build? It sounds like the issues that where fixed in #8308 and #8424 for mouse and keyboard actions respectively.

I do notice that navigating into submenus using keyboard is still borked though.

**Comment by baconpaul** [comment]
Yeah the submenus open automatically on focus requires us to fork juce to fix. it's not easy. see the linked issue to this one.

**Comment by nuoun** [comment]
I mean navigating into a submenu using the arrow keys or space bar does not seem to work at all for the WT menu, the submenu opens but there is no cursor focus.

**Comment by baconpaul** [comment]
Oh right. Well that sounds super bad

**Comment by mkruselj** [comment]
Possibly the custom menu item needs an AccessibilityHandler? Doesn't sound impossible to solve for 1.4, if that's the case.

**Comment by ironcross32** [comment]
I can't say I've ever seen this issue of not being able to navigate into submenus, and I do run the nightlies.

**Comment by nuoun** [comment]
I've been wanting to ask, how do you navigate into the wavetable selection menu @ironcross32? Wonder if the submenus not navigating only happens after opening the wavetable menu by mouse click.

**Comment by ironcross32** [comment]
Right bracket or VO+SHIFT+M on the name of the currently selected wavetable, up and down arrow in the menu, enter or right arrow to dive into a submenu, enter to make the final selection. The process is not that different between Windows and Mac.

> I've been wanting to ask, how do you navigate into the wavetable selection menu [@ironcross32](https://github.com/ironcross32)? Wonder if the submenus not navigating only happens after opening the wavetable menu by mouse click.

**Comment by nuoun** [comment]
Thank you!

So with screenreader on it seems keys do actually does navigate the submenu and it loads the wavetable fine, the menu just misses the focus highlight indicator in the submenus for some reason.

**Comment by baconpaul** [comment]
Do you have cursor mouse tracking on? Also I wonder if it is just the wt menu

I think the problem is cursor mouse tracking opens the menu and focuses it making the keyboard all wonky

**Comment by ironcross32** [comment]
I find the mouse issue an odd one to raise. I wouldn't consider it to be a surge or Juce problem. I've seen it more on Windows, but it can happen on Mac as well. I tend to keep cursor tracking on, but decouple the mouse pointer from the VO cursor.

 What happens is that when you open a context menu, and the mouse pointer just so happens to be within the bounds of one of the menu items that has a submenu attached to it, that submenu will open. For keyboard users, this manifests as keyboard focus jumping inside the submenu as OP has stated. This has been a thing for as long as I've been using computers.

As for the wave table selection bug, this was fixed in #8424 

I have never seen modulations fail to add from the context menu before. However, what might be happening is OP might be unaware you have to type a value in and are navigating away from that typein field and thus, the modulation is never finalized.

@chigkim If you find having the mouse pointer follow the VO cursor helpful for your day-to-day workflows, might I suggest creating a VoiceOver activity for Logic in which that item is disabled?

**Comment by nuoun** [comment]
> I think the problem is cursor mouse tracking opens the menu and focuses it making the keyboard all wonky

Found it, the highlight flag was only set on mouseEnter / mouseExit, should be fixed in #8451

**Comment by chigkim** [comment]
> [@chigkim](https://github.com/chigkim) If you find having the mouse pointer follow the VO cursor helpful for your day-to-day workflows, might I suggest creating a VoiceOver activity for Logic in which that item is disabled?

Yes turning off cursor tracking is a workaround for automatically opening submenus, but many other things fail in Logic as well as Surge synth if you don't have cursor tracking on.

**Comment by chigkim** [comment]
Okay, I just updated to the nightly build, and both issues 1 and 2 are now resolved.
Wavetable now loads correctly, and I was also able to create a modulation by opening the context menu from a target.
Since you guys are tracking the issue with menus expanding automatically in a separate issue, should I close this issue?
Thanks!

**Comment by baconpaul** [comment]
Wonderful! Thanks @nuoun and @chigkim both

and yes  lets use #6582 to track the auto open. I'll close this.
